### ISSUE TITLE
file.AsyncRead doesn't work in the Menu State.

### ISSUE THREAD

**Issue by RaphaelIT7** [Facepunch/garrysmod-issues#5433]
### Details
file.AsyncRead doesn't work in the Menu State. Tested on all Branches

### Steps to reproduce
```lua
file.Write("example.txt", "Example") -- creating a file for testing
file.AsyncRead("example.txt", "DATA", function(_, _, status, content)
	print("Async Read: " .. status, content)
end)
print("Read: " .. file.Read("example.txt", "DATA"))
```
Output:
```lua
Read: Example
```

**Comment by PrikolMen** [comment]
Yes, I have encountered this problem, without the help of external modules it is impossible to fix it.

**Comment by robotboy655** [comment]
I am not interested in any menu state Lua issues. It is not meant to be used at all.

**Comment by PrikolMen** [comment]
indeed

**Comment by RaphaelIT7** [comment]
I looked a bit into it, and internally it seems to work fine as it loads the file and takes up the memory for it, but it never frees it.
I noticed that the `GarrysMod::Lua::Libraries::File::AsyncCycle` function is missing, so I'm guessing that this is the reason why it doesn't call the callback and free the memory.

The menu state isn't meant to be used, but some might want to do something in it and file.AsyncRead can become really useful.
